src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -62,24 +65,45 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
                 {
                     // Process bypass list for manual setting.
                     string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>();
 
                     foreach (string value in list)
                     {
                         string tmp = value.Trim();
+                        if (tmp.Length == 0)
+                        {
+                            continue;
+                        }
                         if (tmp == "<local>")
                         {
                             _bypassLocal = true;
                             continue;
                         }
-                        if (tmp.Length > 0)
+                        try
+                        {
+                            Regex re = new Regex(tmp.Replace(".", "\\.").Replace("*", ".*?") + "$",
```

Comment:
Instead of manually escaping just `.` and `*`, could [`Regex.Escape`](https://msdn.microsoft.com/en-us/library/system.text.regularexpressions.regex.escape(v=vs.110).aspx) be used here to ensure that the pattern is a valid regex when other special characters are present?

Reply:
@nil4 I think the format is not general regex, but rather just the specific prefixes. If that's the case, we should leave it as it is.

Reply:
winhttp supports only *. and that is not regex to start with. This codes converts '*' pattern to regex equivalent. 
BTW the UI does not let you add anything with special characters.

Reply:
@karelz @wfurt this code constructs regexes from input strings. What I meant is, consider whether manually and selectively escaping just these two characters is enough to prevent the proxy bypass regex list from being constructed incorrectly. 

For instance, is it possible to have IPv6 addresses in the proxy bypass list? If so, the characters `[` and `]` may be present in the input, and when converting that input to a regex, these would be interpreted as a [regex character class](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference). This is just one example, but there are other characters that have special meaning in regular expressions. 

Using `Regex.Escape` guarantees that *any* characters in the input string will be interpreted literally:

> Escapes a minimal set of characters (\, *, +, ?, |, {, [, (,), ^, $,., #, and white space) by replacing them with their escape codes. This instructs the regular expression engine to interpret these characters literally rather than as metacharacters.

Reply:
@nil4 that is a very good point, thanks for catching it!